packages/block-editor/src/components/block-lock/modal.js

```diff
@@ -23,16 +24,33 @@ import useBlockLock from './use-block-lock';
 import useBlockDisplayInformation from '../use-block-display-information';
 import { store as blockEditorStore } from '../../store';
 
+function getTemplateLockValue( lock ) {
+	// Prevents all operations.
+	if ( lock.remove && lock.move ) {
+		return 'all';
+	}
+
+	// Prevents inserting or removing blocks, but allows moving existing blocks.
+	if ( lock.remove && ! lock.move ) {
+		return 'insert';
+	}
+
+	return false;
+}
+
 export default function BlockLockModal( { clientId, onClose } ) {
+	const [ applyTemplateLock, setApplyTemplateLock ] = useState( false );
```

Comment:
By using state here, this is always `false` when opening the modal.

https://user-images.githubusercontent.com/16275880/175302914-09a2a25d-a8a0-41e7-bc7e-a052f917e1ab.mov

Reply:
I thought about defaulting to the `templateLock` attribute value if it's truthy but reverted it last minute. Happy to re-implement it.

Reply:
Having this option off by default plays nicely with recent changes - https://github.com/WordPress/gutenberg/pull/41876#issuecomment-1167626207.

What do you think, Nik?

Reply:
I think we need to reflect the setting applied to the initial state otherwise the users wonder why the checkbox becomes unchecked.

Reply:
Fixed in [8262739](https://github.com/WordPress/gutenberg/pull/41876/commits/8262739ce3a5c383e1d8240c44601cf1bec2e361).